### ISSUE TITLE
Lua: support for likwid_pinThread

### ISSUE THREAD

**Issue by mshiryaev** [RRZE-HPC/likwid#112]
Hello,
is any reason why there is no support for thread pinning in lua likwid module (only process pinning)?
Lua users can utilize 'threads' module and thread pinning can be useful.

**Comment by TomTheBear** [comment]
There is no reason for this. When creating the Lua interface, I only added functions that are required by the "frontend" scripts. For me it is surprising that the threads module does not have its own pinning functionality.

**Comment by mshiryaev** [comment]
@TomTheBear thanks for quick reply and adding new functionality.
I mean 'threads' package from here: https://github.com/torch/threads
Actually I am looking for the way how to pin lua threads to cores and likwid package looks like the only one possible approach for that.

**Comment by TomTheBear** [comment]
I saw this 'threads' package and checked the source which threading solution they are using under the hood and whether it has pinning functionality already. With a threading mechanism like Python's the pinThread function wouldn't make any sense.
I'm happy to read that LIKWID provides a unique feature to the Lua community. The support for pinThread is going to be in the 4.3.0 release (currently only a draft release). Of course you can also use the master branch.
If you had time to test it, can you please give me a short feedback. I havn't tested it for now, I just implemented it analogously to the pinProcess function.

**Comment by mshiryaev** [comment]
@TomTheBear yes, I am going to test pinThread from master branch on my application.
Could you please clarify that point: "With a threading mechanism like Python's the pinThread function wouldn't make any sense" ?
As far as I see lua 'threads' package uses pthread under-hood. Is there any problem with that on lua likwid side?

**Comment by mshiryaev** [comment]
@TomTheBear also could you clarify is there any default pinning when user does "require 'likwid'"?
From what place user should do "require 'likwid'" - from master thread or from child thread?

**Comment by TomTheBear** [comment]
The threads created with the [threading](https://docs.python.org/2/library/threading.html) module of Python are all running on the same CPU due to the global interpreter lock. Consequently, pinning the threads does not really work, it either fails or moves all Python threads to the selected CPU. Python has a [multiprocessing](https://docs.python.org/2/library/multiprocessing.html) module that provides threading but it uses processes under the hood, so sharing data between the processes is not that simple.

Pthreads are totally fine for LIKWID, so the threads package for Lua should work fine.

LIKWID does not do anything automatically, thus `require 'likwid'` just loads the functions into Lua's environment. If you want your master thread/process to run already on a specific CPU, you have to pin it yourself.
Since it's pthreads I assume that a require in the master thread is enough but I'm not sure about this. I havn't used the threads package until now.

**Comment by TomTheBear** [comment]
Did you have a chance to test it?
